### PR TITLE
LPS-206201 Create new column for table DLFileVersion. and DLFileEntry

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
This is the first step of the creation of the field display date (publish date) for documents and media

- LPS-206201 add displayDate to the DLFileVersion table
- LPS-206201 build-service
- LPS-206201 baseline
- LPS-206201 upgrade database
- LPS-206201 baseline

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1407401_8377

**Comment by AliciaGarciaGarcia** [comment]
Hi @adolfopa 
taking into account that this PR is failing I will rebase and send on this same pr the next steps (DLFileEntry) 
sorry for the inconvenience

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by marcogalluzzi — APPROVED** [review]

**Comment by marcogalluzzi** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

File: portal-impl/src/com/liferay/portlet/documentlibrary/service.xml [hunk 1]
```diff
@@ -95,6 +96,14 @@
...
 			<finder-column arrayable-operator="OR" name="folderId" />
 		</finder>
+		<finder name="G_LtD" return-type="Collection">
+			<finder-column name="groupId" />
+			<finder-column comparator="&lt;" name="displayDate" />
+		</finder>
+		<finder name="C_LtD" return-type="Collection">
+			<finder-column name="companyId" />
+			<finder-column comparator="&lt;" name="displayDate" />
+		</finder>
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service.xml:106]: Just a question: why are you creating these finders now? Are you 100% we'll be using these in followup PRs?

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by SylviaLuan** [comment]
Hi @AliciaGarciaGarcia,

There is a failure caused by this pull request.
[LocalFile.PortalDatabaseValidationUpgrade#CompareUpgradedAndFreshDatabaseSchemas7413](https://testray.liferay.com/home/-/testray/case_results/2510971872)

1. Import a db from one of the Affected versions [7.4](https://github.com/liferay/liferay-qa-portal-legacy-ee/blob/master/7.4.13/data-archive/data-archive-portal-mysql.zip) to MySQL
2. Upgrade this database to Master (with DXP profile)
3. tools/portal-tools-db-upgrade-client/db_upgrade.sh
4. Connect Master to a different, empty database schema
5. Start master against blank database schema
6. Compare MBCategory table between upgraded and fresh database

Expected Result:
Tables columns matches between the 2 databases for all versions.

Actual Result:
"friendlyURL" column at MBCategory table default values is different

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade-all

**Comment by AliciaGarciaGarcia** [comment]
Hi @SylviaLuan 
I rebased with master, and tested as you said.

I found no differences between so I will try to launch test again

Thanks!

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
I found the problem.

`Key column 'displayDate' doesn't exist in table: create index IX_92309600 on DLFileVersion (status, displayDate);
`
the new index was trying to be created before the table was modified on the upgrade because the upgrade must be done on the portal-impl module, not on the document library.

I will update this PR with the changes and new LPDs names

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1367401_4947

**Comment by AliciaGarciaGarcia** [comment]
LocalFile.PortalDatabaseValidationUpgrade#CompareUpgradedAndFreshDatabaseSchemas7413
```
Error response from daemon: mount /var/lib/docker/containers/d125e874d8341e11c44e1575d77665689d44069f21986bd60331c116e3954aab/hosts:/var/lib/docker/overlay2/1d05c8a93acce4ad0b16f815135d6c30c04fc8845fecf3c0137de3af795ac860/merged/etc/hosts, flags: 0x5000: no space left on device
     [exec] 
```

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**